src/TitleDetails/TitleDetailsContainer.js

```diff
@@ -120,7 +120,7 @@ const TitleDetailsContainer = ({ location, history, mutator, match }) => {
 
       const piece = getDehydratedPiece(values);
 
-      mutator.orderPieces[mutatorMethod](piece)
+      mutator.orderPieces[mutatorMethod]({ ...piece, titleId })
```

Comment:
since it's a part of piece maybe it should be in initialValues?

Reply:
💯